agent/kafka_handler.go

```diff
@@ -17,6 +33,136 @@ type kafkaHandler struct {
 	ctx   kafkaserver2.ConnectionContext
 }
 
+func detectInvalidTopic(name string) error {
+	// Check if the name is empty
+	if len(name) == 0 {
+		return errors.New("the empty string is not allowed")
+	}
+
+	// Check if the name is "." or ".."
+	if name == "." {
+		return errors.New("'.' is not allowed")
+	}
+	if name == ".." {
+		return errors.New("'..' is not allowed")
+	}
+
+	// Check the length of the name
+	if len(name) > maxNameLength {
+		return fmt.Errorf("the length of '%s' is longer than the max allowed length %d", name, maxNameLength)
+	}
+
+	// Check if the name contains only valid characters
+	if !legalChars.MatchString(name) {
+		return fmt.Errorf("'%s' contains one or more characters other than ASCII alphanumerics, '.', '_' and '-'", name)
```

Comment:
better to just say invalid topic name

Reply:
we are matching it with a regex. Again referring this: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/internals/Topic.java#L47-L60

Reply:
Ok, but be careful not to copy the Java code - that would be copyright infringement